subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/espresso/VisibleCenterGlobalCoordinatesProvider.kt

```diff
@@ -0,0 +1,23 @@
+package com.avito.android.test.espresso
+
+import android.graphics.Rect
+import android.view.View
+import androidx.test.espresso.action.CoordinatesProvider
+
```

Comment:
Not ideal implementation but covers more cases.

Reply:
Have you done fullCheck already?

Reply:
Yep. fastCheck and fullCheck are green.
Some functional tests are failed, but they are flaky.